internal/network/machine_network_cidr.go

```diff
@@ -92,11 +93,36 @@ func VerifyVip(hosts []*models.Host, machineNetworkCidr string, vip string, vipN
 		return errors.Errorf("%s <%s> does not belong to machine-network-cidr <%s>", vipName, vip, machineNetworkCidr)
 	}
 	if !IpInFreeList(hosts, vip, machineNetworkCidr, log) {
-		return errors.Errorf("%s <%s> is already in use in cidr %s", vipName, vip, machineNetworkCidr)
+		msg := fmt.Sprintf("%s <%s> is already in use in cidr %s", vipName, vip, machineNetworkCidr)
+		//In that particular case verify that the machine network range is big enough
+		//to accommodates hosts and vips
+		if !isMachineNetworkCidrBigEnough(hosts, machineNetworkCidr, log) {
+			msg = fmt.Sprintf("%s. The machine network range is too small for the cluster. Please redefine the network.", msg)
+		}
+		return errors.Errorf(msg)
 	}
 	return nil
 }
 
+func isMachineNetworkCidrBigEnough(hosts []*models.Host, machineNetworkCidr string, log logrus.FieldLogger) bool {
+	_, cidr, err := parseCIDR(machineNetworkCidr)
+	if err != nil {
+		log.WithError(err).Errorf("can't parse machine cidr %s", machineNetworkCidr)
+		return true
+	}
+
+	networkPrefix, bits := cidr.Mask.Size()
+	numOfHosts := len(hosts)
+	if numOfHosts == 1 {
+		//allow /31 network prefix and down (the smaller the prefix the larger the range)
+		return networkPrefix <= 31
```

Comment:
IPv6 ?

Reply:
replaced with bits and the minimum network width constants